### PR TITLE
National Cloud Section Updates

### PR DESCRIPTION
**Pull request by negoe**
1. Updated some comments I originally made on Kalyan's PR.
2. Updated App registration experience according to the US Gov.
3. Steps for registration using Powershell a re intentionally missing. Planning to add that in further commits.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -14,7 +14,8 @@ endpoint: AAD v1.0
 
 ## About this sample
 
-This sample shows how to build an ASP.NET MVC web application that uses Azure AD for sign-in using the OpenID Connect protocol. Additionally it also introduces developers to the concept of a [multi-tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps) Azure Active Directory application.
+This sample shows how to build an ASP.NET MVC web application that uses Azure AD for sign-in using the OpenID Connect protocol in Azure Global and Microsoft National Cloud environments. Additionally it also introduces developers to the concept of a [multi-tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps) Azure Active Directory application.
```
Comment (jmprieur) [inline README.md]: and Microsoft National Cloud environments => with a variation to explain how to do it in Microsoft National Cloud environments?

Comment (jmprieur) [reply README.md]: Or even:

This sample shows how to build an ASP.NET MVC web application that uses Azure AD for sign-in using the OpenID Connect protocol. Additionally it also introduces developers to the concepts of:
- a [multi-tenant](https://docs.microsoft.com/en-us/azure/active-directory/develop/single-and-multi-tenant-apps) Azure Active Directory application.
- Azure Global and Microsoft National Cloud environments

File: README.md [hunk 2]
```diff
@@ -178,17 +179,30 @@ If you try to sign-in before the tenant administrator has provisioned the app in
...
- If you using automation provided via Powershell to create your app, you need to change the [Configure.ps1](./AppCreationScripts/Configure.ps1) and [Cleanup.ps1](./AppCreationScripts/Cleanup.ps1) as instructed below to append the `-AzureEnvironmentName` parameter. The details on this parameter and its possible values are listed in [Connect-AzureAD](https://docs.microsoft.com/en-us/powershell/module/azuread/connect-azuread?view=azureadps-2.0).
-
- ```Powershell
- Connect-AzureAD -TenantId $tenantId -AzureEnvironmentName AzureUSGovernment
- ```
-
-1. Then follow the steps outlined in [[Steps 3]](#step-3--configure-the-sample-to-use-your-azure-ad-tenant) above and additionally make the following changes in the `TodoListWebApp\Web.Config` file.
+Microsoft cloud services are also available in three separate national clouds. These national cloud versions are physical and logical network-isolated instances of Microsoft enterprise cloud services that are confined within the geographic borders of specific countries and operated by local personnel.
```
Comment (jmprieur) [inline README.md]: In the Overview at the top of the document, Kalyan has done a summary of multi tenancy. I wonder if we would not want to have a summary of sovereign clouds as well there (2 sections)
lines 182-186

File: README.md [hunk 3]
```diff
@@ -178,17 +179,30 @@ If you try to sign-in before the tenant administrator has provisioned the app in
...
+- Microsoft Cloud Germany
+- Azure and Office 365 operated by 21Vianet in China
+
+In order to run this sample on a National Cloud, you can follow the same steps as of Global cloud with a few variations:
+
+#### Step 1: 
+Follow [[Step 1]](#Clone-or-download-this-repository)
+
+#### Step 2: 
+1. Sign in to the Azure Government portal https://portal.azure.us.
```
Comment (jmprieur) [inline README.md]: You particularized the Azure gov here? I wonder if we would not want to have a sub section directly in step 2. Something like:

> If you want to create an app for Microsoft national or sovereign clouds, sign-in to the portal corresponding to that cloud:
  - https://portal.azure.us for the Azure Government
 - See [link to the sovereign cloud table]() for others

File: README.md [hunk 4]
```diff
@@ -178,17 +179,30 @@ If you try to sign-in before the tenant administrator has provisioned the app in
...
+6. While still in the Azure Government portal, choose your application, click on Settings and choose Properties.
+7. Find the Application ID value and copy it to the clipboard.
+8. For the App ID URI, enter https://<your_tenant_name>/TodoListWebApp_MT, replacing <your_tenant_name> with the name of your Azure AD tenant.
+9. From the Settings menu, choose Keys and add a key - select a key duration of either 1 year or 2 years. When you save this page, the key value will be displayed, copy and save the value in a safe location - you will need this key later to configure the project in Visual Studio - this key value will not be displayed again, nor retrievable by any other means, so please record it as soon as it is visible from the Azure Government portal.
+ 
+#### Step 3: 
+1. Follow the steps outlined in [[Steps 3]](#step-3--configure-the-sample-to-use-your-azure-ad-tenant) above and additionally make the following changes in the `TodoListWebApp\Web.Config` file.
     - Find the app key `ida:AADInstance` and replace the existing value with the corresponding sign-in endpoint for the national or sovereign cloud you want to target.
```
Comment (jmprieur) [inline README.md]: Again, I think that this might go directly in Step 3.

File: README.md (hunk 1 above)
Comment (kalyankrishna1) [reply README.md]: Deploying to a gov cloud is an orthogonal topic, that we can repeat for nearly all samples. We have a dedicated pages that introduce national clouds, the readme in the bottom references them. I'll  leave this text as is.

File: README.md (hunk 2 above)
Comment (kalyankrishna1) [reply README.md]: Again, note my earlier comment on staying on topic relevant to the sample, the deleted note had a link for the national cloud page. I suggest we start with 
"National cloud are physical and logical network-isolated instances of Microsoft Azure that are confined within the geographic borders of specific countries and operated by local personnel. For more details, visit [link]" 

followed by

"In order to run this sample on a National Cloud, you can follow through the steps above with a few variations:"

File: README.md [hunk 5]
```diff
@@ -178,17 +179,30 @@ If you try to sign-in before the tenant administrator has provisioned the app in
...
- If you using automation provided via Powershell to create your app, you need to change the [Configure.ps1](./AppCreationScripts/Configure.ps1) and [Cleanup.ps1](./AppCreationScripts/Cleanup.ps1) as instructed below to append the `-AzureEnvironmentName` parameter. The details on this parameter and its possible values are listed in [Connect-AzureAD](https://docs.microsoft.com/en-us/powershell/module/azuread/connect-azuread?view=azureadps-2.0).
-
- ```Powershell
- Connect-AzureAD -TenantId $tenantId -AzureEnvironmentName AzureUSGovernment
- ```
-
-1. Then follow the steps outlined in [[Steps 3]](#step-3--configure-the-sample-to-use-your-azure-ad-tenant) above and additionally make the following changes in the `TodoListWebApp\Web.Config` file.
+Microsoft cloud services are also available in three separate national clouds. These national cloud versions are physical and logical network-isolated instances of Microsoft enterprise cloud services that are confined within the geographic borders of specific countries and operated by local personnel.
+Current national clouds include:
```
Comment (kalyankrishna1) [inline README.md]: Remove with a link to the national cloud home page

File: README.md (hunk 3 above)
Comment (kalyankrishna1) [inline README.md]: Can't use the US Gov cloud as an example, the text has to be generic one pointing to the table provided in the national clouds documentation

File: README.md (hunk 4 above)
Comment (kalyankrishna1) [reply README.md]: I think I have a fir idea of what changes need to be made in my PR. I'll do those..

**Comment by jmprieur** [comment]
@negoe 
closing this PR as Kalyan has leveraged these ideas and the REAME.md (and the code is pretty good now)

=> this sample is national cloud enabled :)

Good job  @negoe and @kalyankrishna1

**Pull request closed without merging**